### ISSUE TITLE
Bookmark + Sync plugins: Fix bookmarks sort order

### ISSUE THREAD

**Issue by dandan** [jonatkins/ingress-intel-total-conversion#1013]
__Questions__
1. Am I the only one having issues with the sort order of bookmarks getting lost?
2. Does the solution below seem like a reasonable one?

__Aim__: Display the bookmarks and folders in Alphabetical order.

__Summary__
Currently, when using the Bookmarks plugin along with the Sync plugin, and I sync my bookmarks between two devices, the order of the bookmarks (and folders) gets lost, and everything ends up in random order.  I would much rather have the bookmarks (and folders) sorted in alphabetical order.

__Detailed Problem Description__
I have IITC installed on my laptop and my android tablet, and have the bookmarks and sync pugins installed on both.  When I do the following steps I end up with badly sorted bookmarks:

1. I add some bookmarks on my laptop
2. I manually drag them into an order that I like
3. I sync the bookmarks to my tablet
4. I add a bookmark on the tablet
5. I sync the bookmarks back to my laptop
6. Restart IITC on both

By the end of step 6, my bookmarks on both the laptop and tablet end up being in random order - not alphabetical, and not what I manually dragged them to be.  I find this annoying.

__Detailed Solution__
I can see a few options on how to fix the ordering of bookmarks.  The obvious solution is store and sync the sort order of bookmarks.  I don't like this solution, because I hate manually sorting things (and it doesn't work on my tablet).  I vastly prefer everything to be alphabetical because it's less effort, and that way it matches the ingress app.

So, I think when displaying the Bookmarks list, the folders and bookmarks should just be sorted alphabetically before they are rendered.

**Comment by lablua** [comment]
I'm also encountering the same issue.

**Comment by nexushoratio** [comment]
Can you give specific examples of how the sort order changes?  Preferably limited to 3 or so portals.  You can export the bookmarks to make a backup, then in Ingress, delete portals until you get a small set that demonstrates the problem, export that list and attach it here.  Then use your original backup to restore your list.

Surprisingly, in computer science, sorting is hard.   Settings on the computer could be responsible for different sort orders.

For example:

# Original unsorted file
$ cat t.txt
Cake
Apple
CUcumber
banana

# a natural sorting based on locales
$ LC_ALL=en_US sort t.txt
Apple
banana
Cake
CUcumber

# Old standard sorting based upon byte values of characters
$ LC_ALL=C sort t.txt
Apple
CUcumber
Cake
banana

Start throwing in characters outside the ASCII character set (basically anything outside of a-zA-Z and so on), and you have all sorts of issues.   Different languages, even different countries, have different sorting orders.  And likely by default, your browser probably uses whatever your system default is.  Given two different computers, particularly if they may be of different generations and came up through different upgrade paths, maybe even different operating systems, and you could have different settings, thus different results.

**Comment by dandan** [comment]
@nexushoratio Yep, from what I can see there is no sorting implemented in the code - it's not a character-set issue.  I'm surprised that I'm the first to find it a problem.

I've made a pull request with the sorting (and made it optional so I don't break anybody's setup).  See https://github.com/jonatkins/ingress-intel-total-conversion/pull/1015.

**Comment by Kizuna-Meraki** [comment]
I'm having this issues as well. I'll sort my bookmarks manually. However, when I refresh my browser, the bookmarks go back to their original (old) order and I lose my manual sort.
